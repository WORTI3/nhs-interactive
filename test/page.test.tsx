import Page from "../app/page";
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { load } from "cheerio";

describe('Home page', () => {
  test('rendered component should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  test('should render the component with the correct links & elements', () => {
    const { container } = render(<Page />);
    const html = container.innerHTML;
    const $ = load(html);
  
    expect($('h1').length).toBe(1);
    expect($('h1').text()).toBe('NHSInteractive');
    expect($('h2').length).toBe(0);
    expect($('a').length).toBe(1);
    expect($('svg').length).toBe(1);
    expect($('p').length).toBe(2);
  });
});