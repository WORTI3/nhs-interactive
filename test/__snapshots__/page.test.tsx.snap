// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page rendered component should match snapshot 1`] = `
<div>
  <h1>
    NHS
    <span
      class="ml-4 text-transparent bg-clip-text bg-gradient-to-r from-blue-700 via-blue-500 to-purple-700"
    >
      Interactive
    </span>
  </h1>
  <div
    class="nhsuk-warning-callout"
  >
    <h3
      class="nhsuk-warning-callout__label"
    >
      <span
        role="text"
      >
        <span
          class="nhsuk-u-visually-hidden"
        >
          Important: 
        </span>
        Disclaimer
      </span>
    </h3>
    <p>
      This is a mock service made to look like an NHS website using their frontend toolkit and react components. Information given may also differ from truth. Do not use this for real medical advice or as a source of truth.
    </p>
  </div>
  <div
    class="nhsuk-inset-text"
  >
    <span
      class="nhsuk-u-visually-hidden"
    >
      An Open AI Key must be set in order to search.
    </span>
    <p>
      An Open AI Key must be set in order to search.
    </p>
  </div>
  <div
    class="nhsuk-action-link"
  >
    <a
      class="nhsuk-action-link__link"
      href="/settings"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__arrow-right-circle"
        height="32"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z"
        />
      </svg>
      <span
        class="nhsuk-action-link__text"
      >
        You can change the OpenAI API Key here
      </span>
    </a>
  </div>
</div>
`;
